### ISSUE TITLE
Missing rest operator in object destructuring

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#295]
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

b7220fdbf5a923c64bb6e5bbbfc2ad689ec779eb

**What code were you trying to parse?**

```
const { children, ...props } = this.props
```

**What did you expect to happen?**

`...` operator should be in the AST.

**What happened?**

```js
require('.').parse("const { children, ...props } = this.props").body[0].declarations[0].id.properties[1]
```

```js
{ type: 'Property',
  range: [ 18, 26 ],
  loc: { start: { line: 1, column: 18 }, end: { line: 1, column: 26 } },
  key:
   { type: 'Identifier',
     range: [ 21, 26 ],
     loc: { start: [Object], end: [Object] },
     name: 'props' },
  value:
   { type: 'Identifier',
     range: [ 21, 26 ],
     loc: { start: [Object], end: [Object] },
     name: 'props' },
  computed: false,
  method: false,
  shorthand: true,
  kind: 'init' }
```

**Comment by soda0289** [comment]
Thanks for the report. I think this is a duplicate of https://github.com/eslint/typescript-eslint-parser/issues/276 and fixed in PR https://github.com/eslint/typescript-eslint-parser/pull/279

Can you try out the master branch?

**Comment by vjeux** [comment]
This is indeed fixed in master, thanks! https://github.com/prettier/prettier/pull/1738
